### PR TITLE
Issue1047 wcag more semantic headings

### PR DESCRIPTION
**Pull request by kouralex**
This PR adds some accessibility headings in order to address issues #1036 #1037 #1038 #1039 #1047 

There are still some things left to be done. For example, this PR is missing translations for some translatable strings that were added (and some of them require a bit of tweaking anyway) and there could be better accessibility advises in many places, e.g., converting every `->` (arrows) into speech.

Additionally, fixes some small issues that were introduced by #1055

### CODE REVIEW

File: view/light.twig [hunk 1]
```diff
@@ -37,43 +37,79 @@
...
             <li id="changes"><a href="{{ request.vocabid }}/{{ request.lang }}/new{% if request.contentLang != request.lang %}?clang={{ request.contentLang }}{% endif %}">{% trans "Changes-nav" %}</a></li>
           {% endif %}
         {% else %}
+          <h3 class="sr-only">{% trans "Search options tab" %}</h3>
           <li id="limit"><p>{% trans "Search options" %}</p></li>
         {% endif %}
         </ul>
       </div>
       {% if letters %}
+        <h4 class="sr-only">{% trans "Choose alphabetical listing letter" %}</h4>
```
Comment (joelit) [inline view/light.twig:75]: As a side note - it is a shame that the screen reader annotations like these only work when traversing the headers on the page with a H-key, but not when traversing clickable items with a Tab-key. However, I really think this is an issue with the screen reader, not with the Skosmos side bar's structure.

**Review by joelit — CHANGES_REQUESTED** [review]
Some tweaks to be done, as mentioned. This is a monumental improvement for the heading structure and makes it feasible to use the h-key with a screen reader.

File: resource/js/docready.js [hunk 2]
```diff
@@ -97,15 +97,32 @@ $(function() { // DOCUMENT READY
       });
     });
 
+    var active_tab = $('li.active').attr('id');
```
[the remark this answers is not in the document]
Comment (kouralex) [reply resource/js/docready.js:100]: That is a bit strange. I am unable to reproduce such behavior - maybe I do not have the right tools on my hand (testing everything with Windows Narrator + Edge). The reason why those two are handled differently is based on the fact that their contents are generated on behalf of a jsTree object, which complicates things. To fix some issues regarding this one should do a more thoroughly overhaul of that code base.

But please, as I am unable to reproduce this, I am asking you to subsequently raising on issue on this matter as I am a bit forced to merge this PR anyhow now.

File: view/search-result.twig [hunk 3]
```diff
@@ -21,7 +21,7 @@
     <a class="prefLabel conceptlabel" href="{% if "isothes:ConceptGroup" in concept.type %}{{ concept.uri | link_url(concept.vocab, request.lang, 'page') }}{% elseif concept.exvocab is defined%}{{ concept.uri | link_url(concept.exvocab,concept.contentLang) }}{% else %}{{ concept.uri | link_url(concept.vocab,request.lang,'page',concept.contentLang) }}{% endif %}">{{ concept.label(request.contentLang) }}</a>{% if explicit_langcodes or request.queryparam('anylang') %}<span class="versal"> ({{ concept.contentLang }})</span>{% endif %}
     {% endif %}
   {% endspaceless %}
-  {% if concept.type == 'skosext:DeprecatedConcept' %}<h2 class="deprecated-alert">{% trans %}deprecated{% endtrans %}</h2>{% endif %}
+  {%- if concept.type == 'skosext:DeprecatedConcept' %}<span class="versal deprecated-alert">{% trans %}deprecated{% endtrans %}</span>{% endif -%}
```
[the remark this answers is not in the document]
Comment (kouralex) [reply view/search-result.twig:24]: Yes, it was intentional as I wanted them to follow the correct heading structure (that is, no h2 inside h3 for example). There might be cases where this should be reverted though, I suggest raising issues as one encounters them. Anyway, the heading level should be assessed with care.

File: view/light.twig (hunk 1 above)
Comment (kouralex) [reply view/light.twig:75]: Yes. There are ways to make this work, e.g., using aria-labelledBy or similar, I think. One solution could be changing the element structure, too. I am not an expert on this subject so I would like to learn more of the best practices how to implement such features. One thing to consider is compatibility - ARIAs may not be fully supported by every reader?

**Comment by kouralex** [comment]
Even though there are some concerns (rightfully) raised by @joelit (btw thank you for a very throughout testing!) I feel like the easiest and especially fastest way to get things on the right tracks now is to merge this and file new issues on any defects pointed out above.

**Pull request merged**